WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ParameterRepository.kt

```diff
@@ -0,0 +1,33 @@
+package com.woocommerce.android.ui.products
+
+import com.woocommerce.android.tools.SelectedSite
+import com.woocommerce.android.ui.products.models.SiteParameters
+import com.woocommerce.android.viewmodel.SavedStateWithArgs
+import org.wordpress.android.fluxc.store.WooCommerceStore
+import javax.inject.Inject
+
+class ParameterRepository @Inject constructor(
+    private val wooCommerceStore: WooCommerceStore,
+    private val selectedSite: SelectedSite
+) {
+    fun getParameters(key: String, savedState: SavedStateWithArgs): SiteParameters {
+        val parameters = savedState.get<SiteParameters>(key) ?: loadParameters()
+        savedState[key] = parameters
+        return parameters
+    }
+
+    private fun loadParameters(): SiteParameters {
+        val currencyCode = wooCommerceStore.getSiteSettings(selectedSite.get())?.currencyCode
+        val gmtOffset = selectedSite.get().timezone?.toFloat() ?: 0f
+        val (weightUnit, dimensionUnit) = wooCommerceStore.getProductSettings(selectedSite.get())?.let { settings ->
+            return@let Pair(settings.weightUnit, settings.dimensionUnit)
+        } ?: Pair(null, null)
```

Comment:
Since the weight and dimension Units can be nullable in case there are no settings, I think the Elvis operator with the optional `let` may be redundant here, we can just place the let to run on a nullable `WCProductSettingsModel?` and fill it with optional values directly. Maybe something like that:
```suggestion
        val (weightUnit, dimensionUnit) = 
            wooCommerceStore
            .getProductSettings(selectedSite.get())
            .let { Pair(it?.weightUnit, it?.dimensionUnit) }
```

Does this make sense?

Reply:
Sure. I've updated the code.